LibertyDSNP/spec
# Features/spellcheck linting

Problem
=======
We need spellchecking and a markdown linter.
[fixes #176963069](https://www.pivotaltracker.com/story/show/12345678)

markdown-spellcheck uses a `.spelling` file that contains custom words for your repo. You can add new words by editing it, adding one word (case sensitive) per line, or by running the new script, `npm run spellfix`

NOTE: I did not include a "lint-fix" equivalent because remark does not have a markdown code rewrite option without adding another package and writing a new script.

Solution
========
1. Add remark-lint, remark and markdown-spellcheck
2. Add scripts that perform linting and spellchecking, including an interactive one.
3. Use in the "lint" package.json script.
Correct any warnings/errors that come from running lint.

Steps to Verify:
----------------
```
npm i
npm run lint
```
The packages should install without dependency conflicts, the spellchecker should run, and the markdown linter should run with no errors or warnings.

You can also test the interactive spellchecking, which allows you to add items to the dictionary, the `.spelling` file, by misspelling something and then running
`npm run spellfix`

Use up-down arrow to select a fix type, such as adding to the dictionary or entering a new spelling.  It should fix the spelling for you or add a new word to the dictionary (for example).
